cli/azd/pkg/auth/manager.go

```diff
@@ -0,0 +1,151 @@
+package auth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/public"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+)
+
+// TODO(azure/azure-dev#710): Right now, we re-use the App Id of the `az` CLI, until we have our own.
+//
+// nolint:lll
+// https://github.com/Azure/azure-cli/blob/azure-cli-2.41.0/src/azure-cli-core/azure/cli/core/auth/identity.py#L23
+const cAZD_CLIENT_ID = "04b07795-8ddb-461a-bbee-02f9e1bf7b46"
+
+// The scopes to request when acquiring our token during the login flow.
+var cLoginScopes = []string{"https://management.azure.com/.default"}
```

Comment:
do we need graph.microsoft.com/.default?

Reply:
We don't seem to?  I was able to run the pipeline config stuff without requesting this scope, and it seemed like including it caused problems when trying to log in with a service principal.  Do you have thoughts on other things to try?